src/simple_minreq.rs

```diff
@@ -0,0 +1,271 @@
+    fn source(&self) -> Option<&(dyn error::Error + 'static)> {
+        use self::Error::*;
+
+        match *self {
+            Json(ref e) => Some(e),
+            Minreq(ref e) => Some(e),
+        }
+    }
+}
+
+impl From<serde_json::Error> for Error {
+    fn from(e: serde_json::Error) -> Self {
+        Error::Json(e)
+    }
+}
+
+impl From<minreq::Error> for Error {
+    fn from(e: minreq::Error) -> Self {
+        Error::Minreq(e)
+    }
+}
+
+impl From<Error> for crate::Error {
+    fn from(e: Error) -> crate::Error {
+        match e {
+            Error::Json(e) => crate::Error::Json(e),
+            e => crate::Error::Transport(Box::new(e)),
+        }
+    }
+}
+
+impl Transport for SimpleHttpTransport {
+    fn send_request(&self, req: Request) -> Result<Response, crate::Error> {
+        Ok(self.request(req)?)
+    }
+
+    fn send_batch(&self, reqs: &[Request]) -> Result<Vec<Response>, crate::Error> {
+        Ok(self.request(reqs)?)
+    }
+
+    fn fmt_target(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.url)
+    }
+}
+
+/// Builder for simple bitcoind [`SimpleHttpTransport`].
+#[derive(Clone, Debug)]
+pub struct Builder {
+    tp: SimpleHttpTransport,
+}
+
+impl Builder {
+    /// Constructs a new [`Builder`] with default configuration and the URL to use.
+    pub fn new() -> Builder {
+        Builder {
+            tp: SimpleHttpTransport::new(),
+        }
+    }
+
+    /// Sets the timeout after which requests will abort if they aren't finished.
+    pub fn timeout(mut self, timeout: Duration) -> Self {
+        self.tp.timeout = timeout;
+        self
+    }
+
+    /// Sets the URL of the server to the transport.
+    pub fn url(mut self, url: &str) -> Result<Self, Error> {
+        self.tp.url = url.to_owned();
+        Ok(self)
+    }
+
+    /// Adds authentication information to the transport.
+    pub fn basic_auth<S: AsRef<str>>(mut self, user: S, pass: Option<S>) -> Self {
+        let mut auth = user.as_ref().to_owned();
+        auth.push(':');
+        if let Some(ref pass) = pass {
+            auth.push_str(pass.as_ref());
+        }
+        self.tp.basic_auth = Some(format!("Basic {}", &base64::encode(auth.as_bytes())));
+        self
+    }
+
+    /// Adds authentication information to the transport using a cookie string ('user:pass').
+    pub fn cookie_auth<S: AsRef<str>>(mut self, cookie: S) -> Self {
+        self.tp.basic_auth = Some(format!("Basic {}", &base64::encode(cookie.as_ref().as_bytes())));
+        self
+    }
+
+    /// Builds the final [`SimpleHttpTransport`].
+    pub fn build(self) -> SimpleHttpTransport {
+        self.tp
+    }
+}
+
+impl Default for Builder {
+    fn default() -> Self {
+        Builder::new()
+    }
+}
+
+impl crate::Client {
+    /// Creates a new JSON-RPC client using a bare-minimum HTTP transport.
+    pub fn simple_http_minreq(
+        url: &str,
+        user: Option<String>,
+        pass: Option<String>,
```

Comment:
In d20ba2eafcaa3cd5245f5a548cbcbb6d14c13432:

I think we should take `&str` rather than `String` here for both params. (Ideally it could be generic but that sucks because if the user passes `None` then the compiler will complain.)

Reply:
What value do these helper constructors even add? I rekon we should get rid of them from all the modules.

Reply:
Can you confirm that you are ok with my removal of these helper functions mate?

Reply:
I had a quick check to make sure the new code was not non-uniform and only `simlple_http` provides these construtors on `Client` so its the odd one out not `minreq_http`.